src/AxelarGateway.sol

```diff
@@ -128,11 +173,12 @@ abstract contract AxelarGateway is IAxelarGateway, AdminMultisigBase {
             // If token address is no specified, it indicates a request to deploy one.
             bytes32 salt = keccak256(abi.encodePacked(symbol));
             tokenAddress = address(new BurnableMintableCappedERC20{ salt: salt }(name, symbol, decimals, cap));
+            _setTokenType(symbol, TokenType.InternalBurnableFrom);
         } else {
             // If token address is specified, ensure that there is a contact at the specified addressed.
-            require(tokenAddress.code.length != uint256(0), 'NOT_TOKEN');
+            require(tokenAddress.codehash != keccak256(''), 'NOT_TOKEN');
```

Comment:
Create a constant for `keccak256('')` so that it doesn't have to hash it again every time.